server/tests/common/fixture.rs

```diff
@@ -0,0 +1,342 @@
+            self.add_user(user);
+        }
+        for group in &groups {
+            self.add_group(group);
+        }
+        for User { username, groups } in state {
+            for group in groups {
+                self.add_user_to_group(username, group);
+            }
+        }
+    }
+
+    fn add_user(&mut self, user: &String) {
+        post::<CreateUser>(
+            &self.client,
+            &self.token,
+            create_user::Variables {
+                user: create_user::CreateUserInput {
+                    id: user.clone(),
+                    email: format!("{}@lldap.test", user),
+                    avatar: None,
+                    display_name: None,
+                    first_name: None,
+                    last_name: None,
+                },
+            },
+        )
+        .expect("failed to add user");
+        self.users.insert(user.clone());
+    }
+
+    fn add_group(&mut self, group: &String) {
+        let id = post::<CreateGroup>(
+            &self.client,
+            &self.token,
+            create_group::Variables {
+                name: group.clone(),
+            },
+        )
+        .expect("failed to add group")
+        .create_group
+        .id;
+        self.groups.insert(group.clone(), id);
+    }
+
+    fn delete_user(&mut self, user: &String) {
+        post::<DeleteUserQuery>(
+            &self.client,
+            &self.token,
+            delete_user_query::Variables { user: user.clone() },
+        )
+        .expect("failed to delete user");
+        self.users.remove(user);
+    }
+
+    fn delete_group(&mut self, group: &String) {
+        let group_id = self.groups.get(group).unwrap();
+        post::<DeleteGroupQuery>(
+            &self.client,
+            &self.token,
+            delete_group_query::Variables {
+                group_id: *group_id,
+            },
+        )
+        .expect("failed to delete group");
+        self.groups.remove(group);
+    }
+
+    fn add_user_to_group(&mut self, user: &String, group: &String) {
+        let group_id = self.groups.get(group).unwrap();
+        post::<AddUserToGroup>(
+            &self.client,
+            &self.token,
+            add_user_to_group::Variables {
+                user: user.clone(),
+                group: *group_id,
+            },
+        )
+        .expect("failed to add user to group");
+    }
+}
+
+impl Drop for LLDAPFixture {
+    fn drop(&mut self) {
+        let users = self.users.clone();
+        for user in users {
+            self.delete_user(&user);
+        }
+        let groups = self.groups.clone();
+        for group in groups.keys() {
+            self.delete_group(group);
+        }
+        self.child
+            .kill()
+            .map_err(|err| println!("Failed to kill LLDAP: {:?}", err))
+            .ok();
+    }
+}
+
+fn get_database_url() -> String {
+    let url = var(DB_KEY).ok();
+    let url = url.unwrap_or("sqlite://e2e_test.db?mode=rwc".to_string());
```

Comment:
Does that end up in the root or in the test folder?

Reply:
This end up in the root, where the bin is run from during the test